File: cache/cache-annotation-processor/src/test/java/io/koraframework/cache/annotation/processor/testcache/DummyCache22.java
package io.koraframework.cache.annotation.processor.testcache;

import io.koraframework.cache.annotation.Cache;
import io.koraframework.cache.caffeine.CaffeineCache;
import io.koraframework.cache.redis.RedisCache;

import java.math.BigDecimal;

@Cache("dummy22")
public interface DummyCache22 extends RedisCache<DummyCache22.Key, String> {

    record Key(String k1, BigDecimal k2) {}
}

File: cache/cache-symbol-processor/src/test/kotlin/io/koraframework/cache/symbol/processor/testcache/DummyCache22.kt
package io.koraframework.cache.symbol.processor.testcache

import io.koraframework.cache.annotation.Cache
import io.koraframework.cache.redis.RedisCache
import java.math.BigDecimal

@Cache("dummy22")
interface DummyCache22 : RedisCache<DummyCache22.Key, String> {

    data class Key(val k1: String, val k2: BigDecimal)
}
